static/js/protein/chart.ts

```diff
@@ -373,20 +469,45 @@ export function drawProteinInteractionChart(
   });
   const seen = new Set();
   //Removes duplicates from the data
-
   reformattedData = reformattedData.filter((entry) => {
     const duplicate = seen.has(entry.name);
     seen.add(entry.name);
     return !duplicate;
   });
-  const height = GRAPH_HEIGHT_M - MARGIN.top - MARGIN.bottom;
+  //Finds the length of the object array
+  let arrayLength = Object.keys(reformattedData).length;
+  //Formats the graph as per number of entries
```

Comment:
This feels quite hacky, could you try to accomplish drawing graphs that don't have enough data points by having a MIN_HEIGHT for charts and then calculating the scaleBand paddingInner and paddingOuter?

scaleBand documentation: https://observablehq.com/@d3/d3-scaleband

Reply:
That really helped! I did the math and came up with an inner bar padding constant with automates as per the data points! It should be all good now